geps/gep-1494/index.md

```diff
@@ -0,0 +1,59 @@
+# GEP-1494: Auth in Gateway API
+
+* Issue: [#1494](https://github.com/kubernetes-sigs/gateway-api/issues/1494)
+* Status: Provisional
+
+(See status definitions [here](/geps/overview/#gep-states).)
+
+## TLDR
+
+Allow authentication works as a policy that can  be attached to both the Gateway and HTTPRoute.
+
+## Goals
+
+1). In the default policy, it can add an ExtensionFilter that specifies the auth so that Gateway can be attached to.
+2). Individual HTTPRoutes can choose to opt out by specifying an empty extension, or a null extension. 
+3). It can also have disabled authentication.
+
+
+## Non-Goals
+
+1). How authentication is implemented.
+2). No authorization is required in this GEP.
```

Comment:
I wonder if this goal isn't leaking some current approaches of separating authentication and authorisation into completely sets of configuration. This is not necessarily a bad practice but tends to lead to additional requirements on how to export and data (typically user info) from one layer to the other. After all, there's only so much use case for authorisation completely disconnected from authentication.

I can see this separation eventually bumping into requests for defining, e.g., a `principal` entity that "magically" becomes available in an authorisation policy/filter, knowing an authentication step was enforced before. Similarly, authorisation may depend on `jwt` under the assumption that a JSON Web Token has been validated already.

At the very least, it open room for discussing sequencing/dependencies in the enforcement of different configs.

I would recommend checking out Kuadrant's [AuthPolicy](https://docs.kuadrant.io/0.8.0/kuadrant-operator/doc/auth/) and Envoy Gateway's [SecurityPolicy](https://gateway.envoyproxy.io/docs/api/extension_types/#securitypolicy) as a references (prior art). Both see auth as a rather unified domain, I reckon.

Reply:
> I wonder if this goal isn't leaking some current approaches of separating authentication and authorisation into completely sets of configuration. This is not necessarily a bad practice but tends to lead to additional requirements on how to export and data (typically user info) from one layer to the other. After all, there's only so much use case for authorisation completely disconnected from authentication.
> 
> I can see this separation eventually bumping into requests for defining, e.g., a `principal` entity that "magically" becomes available in an authorisation policy/filter, knowing an authentication step was enforced before. Similarly, authorisation may depend on `jwt` under the assumption that a JSON Web Token has been validated already.
> 
> At the very least, it open room for discussing sequencing/dependencies in the enforcement of different configs.
> 
> I would recommend checking out Kuadrant's [AuthPolicy](https://docs.kuadrant.io/0.8.0/kuadrant-operator/doc/auth/) and Envoy Gateway's [SecurityPolicy](https://gateway.envoyproxy.io/docs/api/extension_types/#securitypolicy) as a references (prior art). Both see auth as a rather unified domain, I reckon.

Yes. I think that authentication and authorization are generally regarded as separate entities across various domains.  I personally would want to see such seperation. This separation allows security teams to do better control and enforcement over ACL services in authorization, such as implementing a least privilege policy. 

However, I do realise that this approach may introduce an additional layer of complexity, but I think at the current stage, we are more than just to talk different possiblities than to think of implement these.